src/actions/index.ts

```diff
@@ -0,0 +1,50 @@
+export enum ActionKeys {
+    LOAD = "LOAD",
+    LOADED = "LOADED",
+    TOUCH = "TOUCH",
+    TOUCHED = "TOUCHED",
+}
+
+export interface LoadAction {
+    type: ActionKeys.LOAD;
+    id: string;
+}
+
+export const load = (id: string): LoadAction => ({
+   type: ActionKeys.LOAD,
+   id,
+});
+
+export interface LoadedAction {
+    type: ActionKeys.LOADED;
+    oppName: string;
+    touches: number;
+}
+
+export const loaded = (oppName: string, touches: number): LoadedAction => ({
+    type: ActionKeys.LOADED,
+    oppName,
+    touches,
+});
+
+export interface TouchAction {
```

Comment:
`InitTouch` might be a better name.  Typically every async action will have an `init`, `success` & `failed` action.


If you have a bunch of async actions, creating an `init` & `error` for every one for each one gets really repetitive.  

To ease this in QTR I've created a specific UI reducer, and I dispatch a generic `AsyncInitAction` when I start a request, and then an `ExceptionAction` if an error occurs.

The `AsyncInitAction` sets `blockPage: true`.  The reducer also listens for any of the "success" actions that fire async and will unblock the page when it one fires.  Since it's possible that multiple async action might be active at once, you can assign a KEY or GUID and then track multiple ASYNC actions.